atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/async/AsyncCellLoaderTests.java

```diff
@@ -0,0 +1,97 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+
+public class AsyncCellLoaderTests {
+    private static final String DEFAULT_KEYSPACE = "test";
+    private static final TableReference DEFAULT_TABLE = TableReference.create(Namespace.DEFAULT_NAMESPACE, "foo");
+    // tests are imagined as if the visible data has a timestamp lower than 20 and non visible data has timestamp higher
+    private static final long TIMESTAMP = 20L;
+    private static final Cell NON_VISIBLE_CELL = Cell.create(PtBytes.toBytes(100), PtBytes.toBytes(100));
+    private static final Cell VISIBLE_CELL_1 = Cell.create(PtBytes.toBytes(100), PtBytes.toBytes(200));
+    private static final Cell VISIBLE_CELL_2 = Cell.create(PtBytes.toBytes(100), PtBytes.toBytes(300));
+
+    private CqlClient cqlClient;
+    private final ExecutorService testExecutor = MoreExecutors.newDirectExecutorService();
+
+    @Before
+    public void setUp() {
+        cqlClient = mock(CqlClient.class);
+    }
+
+    @Test
+    public void testNoDataVisible() throws Exception {
+        when(cqlClient.executeQuery(any()))
+                .thenReturn(Futures.immediateFuture(Optional.empty()));
+
+        Map<Cell, Long> request = ImmutableMap.of(NON_VISIBLE_CELL, TIMESTAMP);
+        AsyncCellLoader asyncCellLoader = AsyncCellLoader.create(cqlClient, testExecutor, DEFAULT_KEYSPACE);
+        Map<Cell, Value> result = asyncCellLoader.loadAllWithTimestamp(DEFAULT_TABLE, request).get();
+        assertThat(result.isEmpty()).isTrue();
+    }
+
+    @Test
+    public void testFilteringNonVisible() throws Exception {
+        when(cqlClient.executeQuery(any()))
```

Comment:
I don't really understand why we're relying on a particular ordering of execution?

It should be as easy as just making a get query no?

i.e. I want to be able to see/say

if I get a query of this form, return this etc.

with this, I'm relying on the implementation firing off requests in a particular order

Reply:
Tests are fixed so we don't rely on the order.